src/gdb/ringbuffer.c

```diff
@@ -0,0 +1,41 @@
+#include <sof/gdb/ringbuffer.h>
+
+volatile struct ring * const rx = (void *) 0x9e008000;
+volatile struct ring * const tx = (void *) 0x9e008120;
+volatile struct ring * const debug = (void *) 0x9e008220;
```

Comment:
These addresses should be defined in memory.h

Reply:
@lgirdwood, they belong to GDB and only to GDB - meaning they will not be used by any other component other than GDB. So does it really make sense to move it outside GDB ? If a month from now we decided to load GDB on demand these addresses in memory.h will must be again taken away or will become a residue.

Reply:
ok, so how do these addresses align with the xtensa memory map ? If they are special debug only addresses should they not be in core-isa.h ? The intention for memory.h is to be like core-isa.h and allow differentiation per platform and architecture, so this would break on non xtensa architectures.

Reply:
I rethinked this case once more and we have two choices in my opinion - either to keep these addresses where they are now with `#if defined` for different platform (because current version has only debug addresses for ICL) or move them to each platform memory.h file with `#if defined CONFIG_GDB_DEBUG` (we can even skip this precompiler check with no harm). The advantage of having it defined inside GDB rather than memory.h is: less files to maintain, better encapsulation and modular approach.